wandb/sdk/wandb_init.py

```diff
@@ -107,27 +108,33 @@ def setup(self, kwargs) -> None:
         _set_logger(self._wl._get_logger())
 
         # Start with settings from wandb library singleton
-        settings: Settings = self._wl._clone_settings()
+        settings: Settings = self._wl.settings.copy()
         settings_param = kwargs.pop("settings", None)
-        if settings_param:
-            settings._apply_settings(settings_param)
-
-        self._reporter = reporting.setup_reporter(
-            settings=settings.duplicate().freeze()
-        )
-
-        sm_config: Dict = (
-            {} if settings.sagemaker_disable else sagemaker.parse_sm_config()
+        if settings_param is not None:
+            if isinstance(settings_param, Settings):
+                # todo: check the logic here. this _only_ comes up in tests?
```

Comment:
did you resolve this TODO?

Reply:
Ah, I want my precious reviewers to double-check the logic re policy/non-policy settings and their source priorities pleeease :)

Reply:
Ah, haha... task for the reader ;)